src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import cx from "classnames";
+import React, { FC, ReactElement, useEffect, useState } from "react";
+import { SystemTheme, Theme, ThemeColor } from "./ThemeProviderConstants";
+import { generateThemeCssOverride } from "./ThemeProviderUtils";
+import { withStaticProps } from "../../types";
+
+export interface ThemeProviderProps {
+  theme?: Theme;
+  children: ReactElement;
+}
+
+const ThemeProvider: FC<ThemeProviderProps> & { systemThemes?: typeof SystemTheme; colors?: typeof ThemeColor } = ({
+  theme,
+  children
+}) => {
+  const [stylesLoaded, setStylesLoaded] = useState(false);
+
+  useEffect(() => {
+    if (!theme) {
+      return;
+    }
+
+    // Create a new style element
+    const styleElement = document.createElement("style");
+    styleElement.type = "text/css";
+    styleElement.id = theme.name;
+    const themeCssOverride = generateThemeCssOverride(theme);
+
+    try {
+      // Set the innerText property of the style element with the dynamic CSS
+      styleElement.appendChild(document.createTextNode(themeCssOverride));
+
+      // Append the style element to the document head
+      document.head.appendChild(styleElement);
+      setStylesLoaded(true);
+    } catch (error) {
+      console.error("Error loading dynamic CSS:", error);
```

Comment:
that's actually not loading dynamic CSS but rather generating a theme.
And maybe we can add some prefix like `"vibe: "` or something similar so we can monitor such errors if they get caught sentry